grpc/src/main/java/com/linecorp/armeria/common/grpc/GrpcJsonMarshaller.java

```diff
@@ -39,19 +39,62 @@
 
     /**
      * Returns a newly-created {@link GrpcJsonMarshaller} which serializes and deserializes a {@link Message}
-     * served by the {@linkplain ServiceDescriptor service}.
+     * served by the {@linkplain ServiceDescriptor service}. This implementation internally uses
+     * <a href="https://github.com/curioswitch/protobuf-jackson">protobuf-jackson</a>
+     * to serialize and deserialize messages.
      */
     static GrpcJsonMarshaller of(ServiceDescriptor serviceDescriptor) {
-        return builder().build(serviceDescriptor);
+        return ofJackson(serviceDescriptor);
     }
 
     /**
-     * Returns a new {@link GrpcJsonMarshallerBuilder}.
+     * Returns a new {@link GrpcJsonMarshallerBuilder}. This implementation internally uses
+     * <a href="https://github.com/curioswitch/protobuf-jackson">protobuf-jackson</a>
+     * to serialize and deserialize messages.
      */
     static GrpcJsonMarshallerBuilder builder() {
+        return builderForJackson();
+    }
+
+    /**
+     * Returns a newly-created {@link GrpcJsonMarshaller} which serializes and deserializes a {@link Message}
+     * served by the {@linkplain ServiceDescriptor service}. This implementation internally uses
+     * <a href="https://github.com/curioswitch/protobuf-jackson">protobuf-jackson</a>
+     * to serialize and deserialize messages.
+     */
+    static GrpcJsonMarshaller ofJackson(ServiceDescriptor serviceDescriptor) {
+        return builderForJackson().build(serviceDescriptor);
+    }
+
+    /**
+     * Returns a new {@link GrpcJsonMarshallerBuilder}.  This implementation internally uses
+     * <a href="https://github.com/curioswitch/protobuf-jackson">protobuf-jackson</a>
+     * to serialize and deserialize messages.
+     */
+    static GrpcJsonMarshallerBuilder builderForJackson() {
         return new GrpcJsonMarshallerBuilder();
     }
 
+    /**
+     * Returns a newly-created {@link GrpcJsonMarshaller} which serializes and deserializes {@link Message}.
```

Comment:
Should add notes that the performance is much lower and should only be used if the default has problems, for example enum values in proto2 files. In practical terms, `ofGson` is not fast enough for production workloads - actually I don't know if having these factories solves the user's reported issue since it still requires configuration, as opposed to configuring the serialization formats. AFAICT, the user didn't need JSON, but they were having issues because the default serialization formats includes it. Requiring a different option may not actually help them.

Reply:
> Should add notes that the performance is much lower

Thanks for looking though this PR! Left an additional note that addresses this.

> AFAICT, the user didn't need JSON, but they were having issues because the default serialization formats includes it.

I still think this PR has meaning in that users who are committed to using `armeria` still have an option to use the upstream marshaller implementation if they'd like.

Having said this, if we want to avoid breaking changes and avoid user-side extra configuration, I guess the other option we have is to just swallow the exception and warn when building a `MessageMarshaller` fails. Let me also consider this.